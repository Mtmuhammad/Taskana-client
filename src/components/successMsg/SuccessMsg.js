// Component to show success msg for CRUD operations

const SuccessMsg = ({ success, setSuccess }) => {
  return (
    <div
      data-testid="success-container"
      className="msg d-flex justify-content-between alert alert-success alert-dismissible fade show"
      role="alert"
    >
      <p
        data-testid="success-msg"
        style={{ fontSize: "1.3rem", marginBottom: "0" }}
      >
        <strong>{success}</strong>
      </p>
      <i
        data-testid="success-msg-icon"
        type="button"
        className="close text-success bx bx-x"
        data-dismiss="alert"
        aria-label="Close"
        style={{ fontSize: "2.3rem" }}
        onClick={() => setSuccess("")}
      ></i>
    </div>
  );
};

export default SuccessMsg;
