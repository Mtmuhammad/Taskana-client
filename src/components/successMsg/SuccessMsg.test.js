import React from "react";
import { render, screen } from "@testing-library/react";
import SuccessMsg from "./SuccessMsg";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(<SuccessMsg />);
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<SuccessMsg />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("error msg", () => {
  test("should render error msg container", () => {
    render(<SuccessMsg />);
    const container = screen.getByTestId("success-container");
    expect(container).toBeInTheDocument();
    expect(container).toHaveClass(
      "msg d-flex justify-content-between alert alert-success alert-dismissible fade show"
    );
  });
  test("should render error msg", () => {
    render(<SuccessMsg />);
    const msg = screen.getByTestId("success-msg");
    expect(msg).toBeInTheDocument();
  });
  test("should render error msg icon", () => {
    render(<SuccessMsg />);
    const icon = screen.getByTestId("success-msg-icon");
    expect(icon).toBeInTheDocument();
    expect(icon).toHaveClass("close text-success bx bx-x");
  });
});
