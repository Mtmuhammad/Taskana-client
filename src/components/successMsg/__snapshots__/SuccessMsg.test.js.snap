// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="msg d-flex justify-content-between alert alert-success alert-dismissible fade show"
    data-testid="success-container"
    role="alert"
  >
    <p
      data-testid="success-msg"
      style="font-size: 1.3rem; margin-bottom: 0px;"
    >
      <strong />
    </p>
    <i
      aria-label="Close"
      class="close text-success bx bx-x"
      data-dismiss="alert"
      data-testid="success-msg-icon"
      style="font-size: 2.3rem;"
      type="button"
    />
  </div>
</DocumentFragment>
`;
